velox/expression/PrestoCastHooks.cpp

```diff
@@ -65,6 +70,50 @@ Expected<int32_t> PrestoCastHooks::castStringToDate(
   return util::castFromDateString(dateString, util::ParseMode::kStandardCast);
 }
 
+namespace {
+
+using double_conversion::StringToDoubleConverter;
+
+template <typename T>
+Expected<T> doCastToFloatingPoint(const StringView& data) {
+  static const T kNan = std::numeric_limits<T>::quiet_NaN();
+  static StringToDoubleConverter stringToDoubleConverter{
+      StringToDoubleConverter::ALLOW_TRAILING_SPACES,
+      /*empty_string_value*/ kNan,
+      /*junk_string_value*/ kNan,
+      "Infinity",
+      "NaN"};
+  int processedCharactersCount;
+  T result;
+  auto* begin = std::find_if_not(data.begin(), data.end(), [](char c) {
```

Comment:
Would it make sense to use isAsciiWhiteSpace from velox/functions/lib/string/StringImpl.h ?